packages/truffle-core/lib/commands/config.js

```diff
@@ -12,27 +117,94 @@ const command = {
         option: "--disable-analytics",
         description:
           "Disable Truffle's ability to send usage data to Google Analytics"
+      },
+      {
+        option: "enable",
+        discription: "Enable a Truffle Attribute"
+      },
+      {
+        option: "disable",
+        discription: "Disable a Truffle Attribute"
+      },
+      {
+        option: "get",
+        discription: "Get a Truffle Attribute"
+      },
+      {
+        option: "read",
+        discription: "Get a Truffle Attribute"
+      },
+      {
+        option: "set",
+        discription: "Set a Truffle Attribute"
+      },
+      {
+        option: "write",
+        discription: "Set a Truffle Attribute"
       }
     ]
   },
-  builder: {},
+  userLevelSettings: ["analytics"],
+  builder: {
+    _: {
+      type: "string"
+    }
+  },
   /**
-   * run config commands to set user-level config settings
+   * run config commands to get/set project/user-level config settings
    * @param {Object} options
    * @param {Func} callback
    */
   run: function(options, done) {
     const googleAnalytics = require("../services/analytics/google.js");
-    let setAnalytics;
-    if (options.enableAnalytics) {
-      setAnalytics = googleAnalytics.setAnalytics(true);
-      done();
-    } else if (options.disableAnalytics) {
-      setAnalytics = googleAnalytics.setAnalytics(false);
-      done();
+    const Config = require("truffle-config");
+
+    let command;
+    if (options.enableAnalytics || options.disableAnalytics) {
+      // TODO: Deprecate the --(en|dis)able-analytics flag in favor for `enable analytics`
+      command = {
+        set: true,
+        userLevel: true,
+        key: "analytics",
+        value: options.enableAnalytics || false
+      };
     } else {
-      setAnalytics = googleAnalytics.setUserConfigViaPrompt();
+      command = parse(options._);
+    }
+
+    if (command === null) {
+      const setAnalytics = googleAnalytics.setUserConfigViaPrompt();
       setAnalytics.then(() => done()).catch(err => err);
+    } else if (command.userLevel) {
+      switch (command.key) {
+        case "analytics": {
+          if (command.set) {
+            googleAnalytics.setAnalytics(command.value);
+          } else {
+            options.logger.log(googleAnalytics.getAnalytics());
+          }
+          break;
+        }
+      }
+
+      done();
+    } else {
+      Promise.resolve()
+        .then(async () => {
```

Comment:
This doesn't need to be `async`? Unless we just want to do it on principle?

Reply:
No it doesn't anymore. I had copied this chunk from `commands/debug.js` but then removed the `await Environment.detect(...)` without removing the `async`

Reply:
Fixed: https://github.com/trufflesuite/truffle/pull/2302/commits/96a54f2e50ec8e0eafb38dfb1737c6c3562bf314#diff-7deba4495724b2fd4f390796fccbf054R89